internal/controller/apiextensions/claim/reconciler.go

```diff
@@ -378,7 +382,11 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 				return reconcile.Result{Requeue: false}, errors.Wrap(r.client.Status().Update(ctx, cm), errUpdateClaimStatus)
 			}
 
-			if err := r.client.Delete(ctx, cp); resource.IgnoreNotFound(err) != nil {
+			do := &client.DeleteOptions{}
+			if *cm.GetCompositeDeletePolicy() == xpv1.CompositeDeleteForeground {
+				client.PropagationPolicy(metav1.DeletePropagationForeground).ApplyToDelete(do)
```

Comment:
Does this mean the controller is stuck until delete is done? It is very likely that we will run in a timeout with this. It is better to stick with background deletion and check the state in the next reconcile.